add_me_to_your_PATH/Dockerfile.template

```
@@ -7,10 +7,14 @@ FROM julia:${JULIA_VERSION}-buster as pinned-extractor
 
 COPY *Project.toml *Manifest.toml .
 
-# Remove all packages that are not pinned or are not a dependency of a pinned package
+# Remove all packages that are not pinned recursively from the manifest
+# (stdlibs count as pinned)
 RUN julia --project=. -e 'using Pkg;\
-                          pinned = [p.name for p in values(Pkg.dependencies()) if p.is_pinned];\
-                          Pkg.rm([p for p in keys(Pkg.project().dependencies) if p ∉ pinned])'
+                          not_pinned = [p.name for p in values(Pkg.dependencies()) if\
+                                        p.is_direct_dep &&\
+                                        !p.is_pinned &&\
+                                        p.name ∉ values(Pkg.Types.stdlibs())];\
+                          isempty(not_pinned) || Pkg.rm(not_pinned; mode=Pkg.PKGMODE_MANIFEST)'
```

Surface level changes:
```suggestion
                          unpinned = [p.name for p in values(Pkg.dependencies()) \
                                      if p.is_direct_dep && !p.is_pinned && \
                                      p.name ∉ values(Pkg.Types.stdlibs())]; \
                          isempty(unpinned) || Pkg.rm(unpinned; mode=Pkg.PKGMODE_MANIFEST)'
```